tests/deployment_version_test.go

```diff
@@ -192,3 +207,173 @@ func (s *DeploymentVersionSuite) Name() string {
 		farm.Fingerprint32([]byte(fullName)),
 	)
 }
+
+func (s *DeploymentVersionSuite) TestDrainageStatus_SetCurrentVersion_NoOpenWFs() {
```

Comment:
This tests the new parts of the feature (version stops accepting new executions --> child workflow starts --> child sets the status to DRAINING and sleeps for `visibilityGracePeriod` --> child queries visibility, counts 0 open workflows, and sets the status to DRAINED).

It does not test the visibility query, since there are no open workflows, but the visibility query code has not changed from the pre-release, so that should all still work. There are TODOs to test it in the comments below.